src/webapp/context-providers/CurrentOrgUnitContextProvider.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { OrgUnitAccess } from "../../domain/entities/User";
+import { useHistory, useLocation } from "react-router-dom";
+import { CurrentOrgUnitContext, defaultOrgUnitContextState } from "../contexts/current-orgUnit-context";
+import { useAppContext } from "../contexts/app-context";
+
+export const CurrentOrgUnitContextProvider: React.FC = ({ children }) => {
+    const history = useHistory();
+    const location = useLocation();
+    const queryParameters = useMemo(() => new URLSearchParams(location.search), [location]);
+    const orgUnitQueryParam = queryParameters.get("orgUnit");
+
+    const { currentUser } = useAppContext();
+
+    //Set default org unit to the first org unit in list
+    const defaultOrgUnit: OrgUnitAccess = currentUser.userOrgUnitsAccess[0]
+        ? currentUser.userOrgUnitsAccess[0]
+        : defaultOrgUnitContextState.currentOrgUnitAccess;
+    const [currentOrgUnitAccess, setCurrentOrgUnitAccess] = useState<OrgUnitAccess>(defaultOrgUnit);
+
+    const changeCurrentOrgUnitAccess = useCallback(
+        (updatedOrgUnit: string) => {
+            const currentOrgUnitAccess = currentUser.userOrgUnitsAccess.find(ou => ou.orgUnitId === updatedOrgUnit);
+            if (currentOrgUnitAccess) {
+                setCurrentOrgUnitAccess(currentOrgUnitAccess);
+                if (queryParameters.get("orgUnit")) {
+                    queryParameters.set("orgUnit", currentOrgUnitAccess.orgUnitId);
```

Comment:
Same

Reply:
Addressed this by removing queryParameter as a global variable. It is nothing but a intermediate for parsed "location.search". However, did not go ahead with making it an object, because then we would have to write the logic to parse query parameters string ourselves(could be error prone). I thought it might be a good idea to keep using URLSearchParams to parse the query parameters as it is tried and tested. However, Please let me know if you see a downside to this, I will write remove the URLSearchParams and write the parsing logic myself.